example/figure_eight_plan_6dof.py

```diff
@@ -0,0 +1,172 @@
+
+        # Constraint: initial configuration
+        builder.initial_configuration(self.kuka_name, qc[self.kuka.optimized_joint_indexes])
+        builder.initial_configuration(self.kuka_name, time_deriv=1) # initial joint vel is zero
+
+        # Constraint: dynamics
+        builder.integrate_model_states(
+            self.kuka_name,
+            time_deriv=1, # i.e. integrate velocities to positions
+            dt=self.dt,
+        )
+
+        # Get joint trajectory
+        Q = builder.get_robot_states_and_parameters(self.kuka_name)  # ndof-by-T symbolic array for robot trajectory
+
+        # End effector position trajectory
+        pos = self.kuka.get_global_link_position_function(link_ee, n=self.T)
+        pos_ee = pos(Q) # 3-by-T position trajectory for end-effector (FK)
+
+        # Get current position of end-effector
+        pc = self.kuka.get_global_link_position(link_ee, qc)
+        Rc = self.kuka.get_global_link_rotation(link_ee, qc)
+        quatc = self.kuka.get_global_link_quaternion(link_ee, qc)
+
+        # Generate figure-of-eight path for end-effector in end-effector frame
+        path = optas.SX.zeros(3, self.T)
+        path[0, :] = 0.2*optas.sin(t*pi*0.5).T  # need .T since t is col vec
+        path[1, :] = 0.1*optas.sin(t*pi).T  # need .T since t is col vec
+
+        # Put path in global frame
+        for k in range(self.T):
+            path[:,k] = pc + Rc @ path[:,k]
+
+        # Cost: figure eight
+        builder.add_cost_term('ee_path', 1000.*optas.sumsqr(path - pos_ee))
+
+        # Cost: minimize joint velocity
+        dQ = builder.get_robot_states_and_parameters(self.kuka_name, time_deriv=1)
+        builder.add_cost_term('min_join_vel', 0.01*optas.sumsqr(dQ))
+
+        # Prevent rotation in end-effector
+        quat = self.kuka.get_global_link_quaternion_function(link_ee, n=self.T)
+        builder.add_equality_constraint('no_eff_rot', quat(Q), quatc)
+
+        # Setup solver
+        optimization = builder.build()
+        self.solver = optas.CasADiSolver(optimization).setup('ipopt')
+
+    def plan(self, qc):
+
+
+        # Set initial seed, note joint velocity will be set to zero
+        Q0 = optas.diag(qc) @ optas.DM.ones(self.kuka.ndof, self.T)
+        self.solver.reset_initial_seed({f'{self.kuka_name}/q': Q0[self.kuka.optimized_joint_indexes, :]})
+
+        # Set parameters
+        self.solver.reset_parameters({
+            'qc': optas.DM(qc),
+            f'{self.kuka_name}/P': Q0[self.kuka.parameter_joint_indexes, :],
+            f'{self.kuka_name}/dP': optas.DM.zeros(len(self.kuka.parameter_joint_indexes), self.T-1)
+        })
+
+        # Solve problem
+        solution = self.solver.solve()
+
+        # Merge solution with parameterized joint values
+        Q = optas.DM.zeros(self.kuka.ndof, self.T)
+        Q[self.kuka.optimized_joint_indexes, :] = solution[f'{self.kuka_name}/q']
+        Q[self.kuka.parameter_joint_indexes, :] = Q0[self.kuka.parameter_joint_indexes, :]
```

Comment:
This, for me, defeats the point of having parameters. The solution should automatically contain the parametrized joints as well as the optimized joint values since the optimized and parametrized joints are, by definition, part of the same model. This merging should happen automatically in the [`solve`](https://github.com/cmower/optas/blob/01efe251b13b2c615403e28d124da56679f17db0/optas/solver.py#L98-L100) method.

Reply:
This was done this way to avoid major changes that would break the main use case where we optimize all of the joints.
It's not the ideal output but any other way we need to rethink the design of the setting up the optimization variables.

Reply:
I don't think so. As above, I think what you implement here can be put in the `solve` method.

Reply:
As we saw this would require some major changes including changes to some of the nomenclature.
I would stay for know with this work around for this particular case and leave the changes for another PR.
The easier solution would be to include a function in the RobotModel for get_q instead of using the current solution[f'{robot_name}/q'].